File: CHUONG_5/5.22_FragmentState/app/src/main/java/com/example/fragment/MainActivity.java
package com.example.fragment;

import android.os.Bundle;

import androidx.activity.EdgeToEdge;
import androidx.appcompat.app.AppCompatActivity;

import com.example.fragment.databinding.ActivityMainBinding;

public class MainActivity extends AppCompatActivity {
    private ActivityMainBinding binding;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        binding = ActivityMainBinding.inflate(getLayoutInflater());
        super.onCreate(savedInstanceState);
        EdgeToEdge.enable(this);
        setContentView(binding.getRoot());
        if (savedInstanceState == null) {
            getSupportFragmentManager().beginTransaction()
                    .add(R.id.fragmentContainerView, RandomNumberFragment.class, null)
                    .setReorderingAllowed(true)
                    .commit();
        }
    }
}

File: CHUONG_5/5.23(Old)_FragmentManagement/app/src/main/java/com/example/app/MainActivity.kt
package com.example.app

import android.os.Bundle
import android.widget.Toast
import androidx.activity.addCallback
import androidx.activity.enableEdgeToEdge
import androidx.appcompat.app.AppCompatActivity
import androidx.fragment.app.commit
import com.example.app.databinding.ActivityMainBinding

class MainActivity : AppCompatActivity() {
    private lateinit var binding: ActivityMainBinding
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        enableEdgeToEdge()
        binding = ActivityMainBinding.inflate(layoutInflater)
        setContentView(binding.root)

        binding.btnHome.setOnClickListener {
            if (!supportFragmentManager.isStateSaved) {
                supportFragmentManager.commit {
                    setReorderingAllowed(true)
                        .replace(R.id.fragmentContainer, HomeFragment())
                        .addToBackStack(null)
                }
            }
        }

        binding.btnProfile.setOnClickListener {
            if (!supportFragmentManager.isStateSaved) {
                supportFragmentManager.commit {
                    setReorderingAllowed(true)
                    replace(R.id.fragmentContainer, ProfileFragment())
                    addToBackStack(null)
                }
            }
        }

        binding.btnSettings.setOnClickListener {
            if (!supportFragmentManager.isStateSaved) {
                supportFragmentManager.commit {
                    setReorderingAllowed(true)
                    replace(R.id.fragmentContainer, SettingsFragment())
                    addToBackStack(null)
                }
            }
        }

        binding.btnHide.setOnClickListener {
            val currentFragment = supportFragmentManager.findFragmentById(R.id.fragmentContainer)
            if (currentFragment != null) {
                supportFragmentManager.beginTransaction()
                    .hide(currentFragment)
                    .commit()
                Toast.makeText(this, "This fragment is hiding", Toast.LENGTH_LONG).show()
            }
        }

        onBackPressedDispatcher.addCallback(this) {
            if (supportFragmentManager.backStackEntryCount > 0) {
                supportFragmentManager.popBackStack()
            } else {
                finish()
            }
        }
    }
}